src/Workspaces/Core/MSBuild.BuildHost/Program.cs

```diff
@@ -3,34 +3,34 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Collections.Immutable;
 using System.CommandLine;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Rpc;
 using Microsoft.Extensions.Logging;
-
-#if !DOTNET_BUILD_FROM_SOURCE
-using StreamJsonRpc;
-#endif
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Workspaces.MSBuild.BuildHost;
 
 internal static class Program
 {
-#if DOTNET_BUILD_FROM_SOURCE
-
-    internal static void Main()
-    {
-        throw new NotSupportedException("This cannot currently be launched as a process in source build scenarios.");
-    }
-
-#else
-
     internal static async Task Main(string[] args)
     {
+        var propertyOption = new CliOption<string[]>("--property") { Arity = ArgumentArity.ZeroOrMore };
         var binaryLogOption = new CliOption<string?>("--binlog") { Required = false };
-        var command = new CliRootCommand { binaryLogOption };
+        var command = new CliRootCommand { binaryLogOption, propertyOption };
```

Comment:
Random thought. Since this will eventually ship in the SDK, by being used in dotnet-format and dotnet-watch, System.CommandLine will need to be updated to the version used by the SDK being inserted into. In a perfect world they could just patch the version when doing a source build, but the reality is that there are often breaking changes in that library.